pantab/_reader.py

```diff
@@ -1,18 +1,55 @@
 import pathlib
 import shutil
 import tempfile
-from typing import Dict, Union
+from typing import Dict, Optional, Union
 
 import numpy as np
 import pandas as pd
 import tableauhyperapi as tab_api
 
 import libreader  # type: ignore
 import pantab._types as pantab_types
+from pantab._hyper_util import ensure_hyper_process
 
 TableType = Union[str, tab_api.Name, tab_api.TableName]
 
 
+def _read_query_result(
+    result: tab_api.Result, dtypes: Optional[Dict[str, str]],
+) -> pd.DataFrame:
+    if dtypes is None:
+        dtypes = {}
+        # Construct data types from result
+        for column in result.schema.columns:
+            # `result.schema` does not provide nullability information.
```

Comment:
Can you add a note about this in `examples.rst`? Some of the dtype mapping machinery can get complicated here so want to be transparent about it

Reply:
Added this to the documentation in the "Type Mapping" section on the "Usage Notes" page.
Please let me know if that works for you or if you prefer it to be in `examples.rst`